### DISCUSSION TITLE
About the usage of `boolstr` in the config parsing code

### DISCUSSION THREAD

**Discussion by ammgws** [i3/i3#4681] [Q&A] [answered]
For example, let's look at `hide_edge_borders`.

docs:
> **hide_edge_borders** none|vertical|horizontal|both|smart

However looking at the [code](https://github.com/i3/i3/blob/117fb13177824f008c86b827864fa6ebeeccb548/src/config_directives.c#L300-L315), any of the keywords defined in [boolstr](https://github.com/i3/i3/blob/next/libi3/boolstr.c) will set `hide_edge_borders` to `vertical`. This behaviour does not seem to be documented anywhere, and git blame shows it has always been like this.

Is this just an oversight? It makes sense for other commands such as `focus_wrapping` since the docs specifically mention the legacy yes/no options, but seems out of place here.

**Comment by Airblader** [comment, marked as answer]
I think this is compatibility behavior from when we only had a yes/no flag for this. We don't necessarily document that legacy behavior anymore because we don't want to encourage using it, but we'll support it for as long as it doesn't cause any trouble.

**Reply by ammgws** [reply]
Related question about commands like `workspace_auto_back_and_forth` that only have yes/no: these all use boolstr so that means 
1) they accept values apart from `yes` . I guess this isn't documented because the use of boolstr is for making the codebase easier to maintain?
2) any string that isn't in boolstr will be the same as "no". This means the default value is "no", so I guess it should be documented as such?

**Reply by Airblader** [reply]
> I guess this isn't documented because the use of boolstr is for making the codebase easier to maintain?

Accepting other common values is merely a quality of life kind of user experience thing. We could be super strict and error out for any other value, but that's just unnecessary. If a user puts in true instead we still know what they meant.

> This means the default value is "no"

That's not necessarily true, the option might just not be listed in a config at all. Documenting defaults sounds reasonable to me, though.

**Reply by ammgws** [reply]
Yep fair enough, just wanted to know the intent behind it :)
